tests/Model/CollectionInfoTest.php

```diff
@@ -8,19 +8,47 @@
 
 class CollectionInfoTest extends TestCase
 {
-    public function testGetName()
+    public function testGetBasicInformation() : void
     {
-        $info = new CollectionInfo(['name' => 'foo']);
+        $info = new CollectionInfo([
+            'name' => 'foo',
+            'type' => 'view',
+            'options' => ['capped' => true, 'size' => 1048576],
+            'info' => ['readOnly' => true],
+            'idIndex' => ['idIndex' => true], // Dummy option
+        ]);
+
         $this->assertSame('foo', $info->getName());
+        $this->assertSame('foo', $info['name']);
+
+        $this->assertSame('view', $info->getType());
+        $this->assertSame('view', $info['type']);
+
+        $this->assertSame(['capped' => true, 'size' => 1048576], $info->getOptions());
+        $this->assertSame(['capped' => true, 'size' => 1048576], $info['options']);
+
+        $this->assertSame(['readOnly' => true], $info->getInfo());
+        $this->assertSame(['readOnly' => true], $info['info']);
+
+        $this->assertSame(['idIndex' => true], $info->getIdIndex());
+        $this->assertSame(['idIndex' => true], $info['idIndex']);
     }
 
-    public function testGetOptions()
+    public function testMissingFields() : void
     {
-        $info = new CollectionInfo(['name' => 'foo']);
+        $info = new CollectionInfo([
+            'name' => 'foo',
+            'type' => 'view',
+        ]);
+
         $this->assertSame([], $info->getOptions());
+        $this->assertFalse(isset($info['options']));
```

Comment:
FYI: you can use `assertArrayNotHasKey` here for more verbose messages. ArrayHasKey constraint supports ArrayAccess and invokes `offsetExists`.